module-1/app-web/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM python:3.8
+
+# set a directory for the app
+WORKDIR /usr/src/app
+
+# copy all the files to the container
+COPY . .
+
+# install dependencies
+RUN pip install --no-cache-dir -r requirements.txt
```

Comment:
best practice is 

1. to copy `requirements.txt`
2. install them: pip install --no-cache-dir -r requirements.txt
3. copy rest of the code

Why? Docker uses cache a lot, and usually installing requirements.txt takes time, but requirements.txt does not change as often as the code. So every time you copy everything and install requirements.txt, you install them from scratch. In the case above, you just rely on the cache.